File: bin/fibonacci.js
'use strict';
//  Lab: Fibonacci Calculations
//
// Alternating between Ruby and JavaScript, write scripts, in bin/fibonacci.rb and bin/fibonacci.js, that:
//


// calculates the product of the even elements of fibs(excluding zero).





let fibs = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34];

const product = (a,b) => {
  if (a > 0) {
  return a * b;
} else {
  return b * 1;
}
};

let productFibs = fibs.reduce(product, 0);

console.log("The product of the fib numbers is", productFibs);


// sum of all the elemetns of fibs array
const sum = (a,b) => {
  return a + b;
};

let sumFibs = fibs.reduce(sum, 0);

console.log("The sum of the fib numbers is", sumFibs);

// calculates the sum of the odd elements of fibs

const isOdd = function(i) {
  return (i %2 !== 0)
};

let sumIsOdd = fibs.filter(isOdd).reduce(sum, 0);

console.log("The odd sum of the fib numbers is", sumIsOdd);

// calculates the product of elements of fibs (excluding zero).

const isEven = function(i) {
  return (i %2 === 0)
};

let productIsEven = fibs.filter(isEven).reduce(product, 1);
console.log("The even sum of the fib numbers is", productIsEven);

File: bin/fibonacci.rb
# Lab: Fibonacci Calculations
#
# Alternating between Ruby and JavaScript, write scripts, in bin/fibonacci.rb
# and bin/fibonacci.js, that:
#


fibs = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34]

# sum of all the elements of the fibs array

sum_fibs = fibs.reduce(0) { |sum, fib| sum + fib }
puts "The sume of the first ten fibonacci numbers is #{sum_fibs}"

sum_fibs = fibs.reduce(0,:+)
puts "The sume of the first ten fibonacci numbers is #{sum_fibs}"

# product of all fibs  (excluding zero)
#My answer:

product_fibs = fibs[1..-1].reduce(1, :*)
puts "The product of the first ten fibonacci numbers is #{product_fibs}"

#Jeff's answer:

product_fibs = fibs.reject(&:zero?).reduce(:*)


# calculates the sum of the odd elements of fibs
# My answer:

odd_sum_fibs = fibs.reduce(i,:+)

# Jeff's answer
sum_odd = fibs.select(&:odd?).reduce(:+)

# calculates the product of the even elements of fibs(excluding zero).

#Jeff's answer
prod_even = fibs.reject(&:zero?).select(&:even?).reduce(:*)
